src/pyoframe/constants.py

```diff
@@ -6,12 +6,20 @@
 MIT License
 """
 
+import importlib.metadata
+import typing
 from dataclasses import dataclass
 from enum import Enum
-import typing
 from typing import Literal, Optional, Union
+
 import polars as pl
+from packaging import version
 
+# We want to try and support multiple major versions of polars
+try:
+    POLARS_VERSION = version.parse(importlib.metadata.version("polars"))
+except importlib.metadata.PackageNotFoundError:
+    POLARS_VERSION = None
```

Comment:
nit: I would just let it through the error rather than having `POLARS_VERSION = None` don't you think? If polars doesn't exist the entire library doesn't work and if we allow `None` values on the string it means we need to add checks later in the code.

Reply:
I agree!